glide-core/telemetry/src/open_telemetry.rs

```diff
@@ -0,0 +1,240 @@
+use opentelemetry::global::ObjectSafeSpan;
+use opentelemetry::trace::SpanKind;
+use opentelemetry::{global, trace::Tracer};
+use opentelemetry_sdk::propagation::TraceContextPropagator;
+use opentelemetry_sdk::trace::TracerProvider;
+use std::path::PathBuf;
+use std::sync::{Arc, RwLock};
+
+const SPAN_WRITE_LOCK_ERR: &str = "Failed to get span write lock";
+const TRACE_SCOPE: &str = "valkey_glide";
```

Comment:
is not only trace scope, it also metric and logs scopes. maybe OP_SCOPE or something more general is better here